### PR TITLE
[PATCH v6] api/helper/linux-gen: remove deprecated items

### PR DESCRIPTION
**Pull request by TuomasTaipale**
Remove roughly a year old or older deprecated items.

v2:
- Matias' comments
- Fixed checkpatch issue
- Re-introduced `odp_timer_set_t` typedef

v3:
- Added reviewed-by tags

v4:
- Further cleanup due to removal of  `odp_cos_with_l2_priority()` and `odp_cos_with_l3_qos()`

v5:
- Rebased

### CODE REVIEW

File: include/odp/api/spec/classification.h [hunk 1]
```diff
@@ -856,67 +835,6 @@ uint32_t odp_cls_cos_num_queue(odp_cos_t cos);
...
- * @param pktio_in     Ingress port identifier.
- * @param num_qos      Number of QoS levels, typically 8.
- * @param qos_table    Values of the Layer-2 QoS header field.
- * @param cos_table    Class-of-service assigned to each of the allowed
- *                     Layer-2 QOS levels.
- *
- * @retval  0 on success
- * @retval <0 on failure
- */
-int ODP_DEPRECATE(odp_cos_with_l2_priority)(odp_pktio_t pktio_in, uint8_t num_qos,
```
Comment (MatiasElo) [inline include/odp/api/spec/classification.h:899]: `odp_cos_with_l2_priority()` is still mentioned in `doc/users-guide/users-guide-cls.adoc`.

File: helper/include/odp/helper/threads.h [hunk 2]
```diff
@@ -286,39 +276,6 @@ int odph_thread_create(odph_thread_t thread[],
...
- *
- * @param thread_tbl    Thread table
- * @param mask          CPU mask
- * @param thr_params    ODP thread parameters
- *
- * @return Number of threads created
- *
- * @deprecated Use odph_thread_create() instead.
- */
-int ODPH_DEPRECATE(odph_odpthreads_create)(
```
Comment (MatiasElo) [inline helper/include/odp/helper/threads.h:302]: `odph_odpthreads_create()` is still mentioned in `odph_thread_create()` and `odph_thread_param_t` spec.

File: helper/include/odp/helper/threads.h [hunk 3]
```diff
@@ -286,39 +276,6 @@ int odph_thread_create(odph_thread_t thread[],
...
- * Returns when all odpthreads have terminated.
- *
- * @param thread_tbl    Thread table
- * @return The number of joined threads or -1 on error.
- * (error occurs if any of the start_routine return non-zero or if
- *  the thread join/process wait itself failed -e.g. as the result of a kill)
- *
- * @deprecated Use odph_thread_join() instead.
- */
-int ODPH_DEPRECATE(odph_odpthreads_join)(
```
Comment (MatiasElo) [inline helper/include/odp/helper/threads.h:319]: `odph_odpthreads_join()` is still mentioned in `odph_thread_join()` spec.

File: include/odp/api/spec/timer_types.h [hunk 4]
```diff
@@ -548,36 +531,6 @@ typedef enum {
 
 } odp_timer_retval_t;
 
-/**
- * For backwards compatibility, odp_timer_set_t is synonym of odp_timer_retval_t
- *
- * @deprecated Use odp_timer_retval_t instead.
- */
-typedef odp_timer_retval_t odp_timer_set_t;
```
Comment (MatiasElo) [inline include/odp/api/spec/timer_types.h:556]: The deprecation comment was added in June (v1.41.1) and no deprecation macros have been used. Safe to be removed?

File: include/odp/api/spec/classification.h (hunk 1 above)
Comment (TuomasTaipale) [reply include/odp/api/spec/classification.h:899]: V2.

File: helper/include/odp/helper/threads.h (hunk 2 above)
Comment (TuomasTaipale) [reply helper/include/odp/helper/threads.h:302]: V2.

File: helper/include/odp/helper/threads.h (hunk 3 above)
Comment (TuomasTaipale) [reply helper/include/odp/helper/threads.h:319]: V2.

File: include/odp/api/spec/timer_types.h (hunk 4 above)
Comment (TuomasTaipale) [reply include/odp/api/spec/timer_types.h]: Added the typedef back in V2. Will deprecate it in a separate PR.

**Review by MatiasElo — APPROVED** [review]
A period could still be added to the end of `api: sched: remove deprecated items` commit message. Otherwise OK.

**Review by MatiasElo — APPROVED** [review]
Needs rebase.

**Pull request merged**